lib/default.nix

```diff
@@ -9,85 +15,121 @@ let
+      cargoLlvmCov = callPackage ./cargoLlvmCov.nix { };
+      cargoNextest = callPackage ./cargoNextest.nix { };
+      cargoTarpaulin = callPackage ./cargoTarpaulin.nix { };
+      cargoTest = callPackage ./cargoTest.nix { };
+      cleanCargoSource = callPackage ./cleanCargoSource.nix {
+        inherit internalCrateNameForCleanSource;
+      };
+      cleanCargoToml = callPackage ./cleanCargoToml.nix { };
+      configureCargoCommonVarsHook = callPackage ./setupHooks/configureCargoCommonVars.nix { };
+      configureCargoVendoredDepsHook = callPackage ./setupHooks/configureCargoVendoredDeps.nix { };
+      devShell = callPackage ./devShell.nix { };
 
-  crateRegistries = self.registryFromDownloadUrl {
-    dl = "https://static.crates.io/crates";
-    indexUrl = "https://github.com/rust-lang/crates.io-index";
-  };
+      crateNameFromCargoToml = callPackage ./crateNameFromCargoToml.nix {
+        inherit internalCrateNameFromCargoToml;
+      };
 
-  downloadCargoPackage = callPackage ./downloadCargoPackage.nix { };
-  downloadCargoPackageFromGit = callPackage ./downloadCargoPackageFromGit.nix { };
-  filterCargoSources = callPackage ./filterCargoSources.nix { };
-  findCargoFiles = callPackage ./findCargoFiles.nix { };
-  inheritCargoArtifactsHook = callPackage ./setupHooks/inheritCargoArtifacts.nix { };
-  installCargoArtifactsHook = callPackage ./setupHooks/installCargoArtifacts.nix { };
-  installFromCargoBuildLogHook = callPackage ./setupHooks/installFromCargoBuildLog.nix { };
-  mkCargoDerivation = callPackage ./mkCargoDerivation.nix { };
-  mkDummySrc = callPackage ./mkDummySrc.nix { };
+      crateRegistries = self.registryFromDownloadUrl {
+        dl = "https://static.crates.io/crates";
+        indexUrl = "https://github.com/rust-lang/crates.io-index";
+      };
 
-  overrideToolchain = toolchain: self.overrideScope (_final: _prev: {
-    cargo = toolchain;
-    clippy = toolchain;
-    rustc = toolchain;
-    rustfmt = toolchain;
-  });
+      downloadCargoPackage = callPackage ./downloadCargoPackage.nix { };
+      downloadCargoPackageFromGit = callPackage ./downloadCargoPackageFromGit.nix { };
+      filterCargoSources = callPackage ./filterCargoSources.nix { };
+      findCargoFiles = callPackage ./findCargoFiles.nix { };
+      inheritCargoArtifactsHook = callPackage ./setupHooks/inheritCargoArtifacts.nix { };
+      installCargoArtifactsHook = callPackage ./setupHooks/installCargoArtifacts.nix { };
+      installFromCargoBuildLogHook = callPackage ./setupHooks/installFromCargoBuildLog.nix { };
+      mkCargoDerivation = callPackage ./mkCargoDerivation.nix { };
+      mkDummySrc = callPackage ./mkDummySrc.nix { };
 
-  path = callPackage ./path.nix {
-    inherit internalCrateNameForCleanSource;
-  };
+      overrideToolchain = toolchain: self.overrideScope (_final: _prev:
+        let
+          splicedToolchain = spliceToolchain toolchain;
```

Comment:
Manually splice the toolchain using `pkgs.splicePackages` instead of relying on `otherSplices`. I don't think there's any other way if we're using `overrideScope`, because `overrideScope` doesn't update `otherSplices`.